lib/ansible/modules/cloud/docker/docker_service.py

```diff
@@ -728,7 +735,7 @@ def cmd_up(self):
                         ))
                     result['actions'].append(result_action)
 
-        if not self.check_mode and result['changed']:
+        if not self.check_mode and result['changed'] or self.remove_orphans:
```

Comment:
I think this logic is wrong: it should probably be
```suggestion
        if not self.check_mode and (result['changed'] or self.remove_orphans):
```
You don't want this to execute in check mode.

Reply:
With this logic, the `docker_service`/`docker_compose` module will remove the orphans, but it will do so without setting `result['changed']` to `True`, so no change will be detected. This change will introduce another bug.